mlflow/pyfunc/scoring_server/__init__.py

```diff
@@ -82,6 +85,7 @@
 )
 
 
+@deprecated("infer_and_parse_data", "2.4.2")
```

Comment:
Can we add a comment to this method indicating that we're only keeping it in order to maintain compatibility with MLserver?